pennylane_qrack/qrack_device.cpp

```diff
@@ -534,16 +536,19 @@ struct QrackDevice final : public Catalyst::Runtime::QuantumDevice {
     QrackDevice &operator=(QuantumDevice &&) = delete;
 
     auto AllocateQubit() -> QubitIdType override {
-        const QubitIdType label = qubit_map.rbegin()->first + 1U;
-        qubit_map[label] = qsim->Allocate(1U);
+        if (allocated_qubits >= qubit_map.size()) {
+            throw std::runtime_error("Catalyst has requested more qubits than exist in device. (Set your wires count high enough, for the device.)");
```

Comment:
Is there a way to print out this information in the runtime error? That is, instead of saying that the request exceeds the resources, is it possible to put these figures in the output string to give more direction for the user?

Reply:
I considered this, and we do it elsewhere in the device back end, as appropriate. However, in this case, what we're trying to support is the case where the logical qubit wire map input is arbitrary (unsigned integers). As such, we could report the label that failed, but that doesn't tell the user much.

I think it's more helpful to explain to the  user that Catalyst requires in general that they allocate enough logical qubits up-front (David tells me this is a general constraint in Catalyst, for now).

That's good general advice we follow elsewhere, but this is better in this specific case.